### PR TITLE
fleetd: try ports 2379 and 4001 by default for etcd

### PR DESCRIPTION
**Pull request by crawford**
Fixes https://github.com/coreos/fleet/issues/1171.

### CODE REVIEW

File: etcd/client.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ type transport interface {
 
 func NewClient(endpoints []string, transport *http.Transport, actionTimeout time.Duration) (*client, error) {
 	if len(endpoints) == 0 {
-		endpoints = []string{defaultEndpoint}
+		endpoints = strings.Split(defaultEndpoints, ",")
```
Comment (jonboulle) [inline etcd/client.go]: would rather just define defaultEndpoints as a []string

**Comment by jonboulle** [comment]
LGTM, with the (discussed) caveat that it will result in suboptimal performance in scenarios where it isn't listening on both ports. Probably get an ack from @bcwaldon

File: etcd/client.go (hunk 1 above)
Comment (crawford) [inline etcd/client.go]: fixed

**Comment by bcwaldon** [comment]
Could we switch the ordering to prefer 4001 for the time-being? I believe the client is so dumb it will always start at the beginning of the list of endpoints for every request, ignoring the last known healthy node.

**Comment by crawford** [comment]
Done.

**Comment by bcwaldon** [comment]
@crawford Please update `fleet.conf.sample` and `Documentation/deployment-and-configuration.md` with this new behavior. grep for 4001

**Comment by crawford** [comment]
@bcwaldon updated.

File: Documentation/using-the-client.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The `fleetctl` binary is included in all CoreOS distributions, so it is as simpl
 
 ### Custom etcd Endpoint
 
-fleetctl currently requires direct communication with the etcd cluster that your fleet machines are configured to use. Use the `--endpoint` flag to override the default of `http://127.0.0.1:4001`:
+fleetctl currently requires direct communication with the etcd cluster that your fleet machines are configured to use. Use the `--endpoint` flag to override the default of `http://127.0.0.1:2379,http://127.0.0.1:4001`:
```
Comment (bcwaldon) [inline Documentation/using-the-client.md]: The order of the default here is backwards.

File: functional/platform/nspawn.go [hunk 3]
```diff
@@ -243,7 +243,7 @@ func (nc *nspawnCluster) prepFleet(dir, ip, sshKeySrc, fleetdBinSrc string) erro
 	}
 
 	cfgTmpl := `verbosity=2
-etcd_servers=["http://172.17.0.1:4001"]	
+etcd_servers=["http://172.17.0.1:2379"]
```
Comment (bcwaldon) [inline functional/platform/nspawn.go]: Does this work?

**Comment by bcwaldon** [comment]
@crawford so close, crawfish. LGTM after my two comments are addressed

File: Documentation/using-the-client.md (hunk 2 above)
Comment (crawford) [inline Documentation/using-the-client.md]: This is the order that fleetctl uses. (yes, it's opposite of the daemon)

File: functional/platform/nspawn.go (hunk 3 above)
Comment (crawford) [inline functional/platform/nspawn.go]: If the tests pass, isn't assumed that it did work?

File: Documentation/using-the-client.md (hunk 2 above)
Comment (bcwaldon) [inline Documentation/using-the-client.md]: Ok.

File: functional/platform/nspawn.go (hunk 3 above)
Comment (bcwaldon) [inline functional/platform/nspawn.go]: The functional tests don't run in travis.

Comment (bcwaldon) [inline functional/platform/nspawn.go]: Just revert this change for now.

Comment (crawford) [inline functional/platform/nspawn.go]: I see. Reverted.

**Comment by bcwaldon** [comment]
LGTM
